File: src/types/player.ts
import type { Location } from "./location";

export type PlayerStatus = "online" | "offline";
export type GameType = "ets2" | "ats";

export interface LivePosition {
  x: number;
  y: number;
  z?: number;
}

export interface Player {
  id: number;
  truckersmpId: number;
  username: string;
  role: string;
  membership: string;
  avatarUrl?: string;

  status: PlayerStatus;
  game?: GameType;
  server?: string;

  location?: Location;
  livePosition?: LivePosition;
  distance?: string;
}

File: src-tauri/src/models/player.rs
use serde::{Deserialize, Serialize};

use super::LivePosition;

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "lowercase")]
pub enum GameType {
    Ets2,
    Ats,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "lowercase")]
pub enum PlayerStatus {
    Online,
    Offline,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Player {
    pub id: u64,
    pub truckersmp_id: u64,
    pub username: String,
    pub role: String,
    pub membership: String,
    pub avatar_url: Option<String>,

    pub status: PlayerStatus,
    pub game: Option<GameType>,
    pub server: Option<String>,

    pub location: Option<super::Location>,

    pub live_position: Option<LivePosition>,

    pub distance: Option<String>,
}